src/Network/Ethereum/Api/Provider.hs

```diff
@@ -51,22 +56,27 @@ instance Exception Web3Error
 
 --TODO: Change to `HttpProvider ServerUri | IpcProvider FilePath` to support IPC
 -- | Web3 Provider
-data Provider = HttpProvider String
+data Provider = HttpProvider String | WSProvider String Int
   deriving (Show, Eq, Generic)
 
-instance Default Provider where
-  def = HttpProvider "http://localhost:8545"
+defaultHttpPovider :: Provider
+defaultHttpPovider = HttpProvider "http://localhost:8545" -- ^ Default HTTP Provider URI
+
+defaultWSPovider   :: Provider
+defaultWSPovider   = WSProvider   "127.0.0.1" 8546        -- ^ Default WS Provider URI 
 
 -- | 'Web3' monad runner, using the supplied Manager
 runWeb3With :: MonadIO m
             => Manager
             -> Provider
             -> Web3 a
             -> m (Either Web3Error a)
-runWeb3With manager provider f =
-    runWeb3' provider $ jsonRpcManager .= manager >> f
+runWeb3With manager provider f = do  
```

Comment:
@amany9000 Could you please explain this part of code.

Reply:
It gave this error - 

   ```
Couldn't match type ‘a0
                           -> Data.Functor.Identity.Identity Manager’
                     with ‘JsonRpcClient’
      Expected type: Lens.Micro.Type.ASetter
                       JsonRpcClient JsonRpcClient a0 Manager
        Actual type: JsonRpcClient -> Manager
```